### PR TITLE
Add more Knative GSoC 23 ideas

### PR DESCRIPTION
**Pull request by aliok**
Add 3 more Knative GSoC 23 ideas, add additional mentor to an existing idea

Signed-off-by: Ali Ok <aliok@redhat.com>

### CODE REVIEW

**Review by nate-double-u — DISMISSED** [review]
Thanks for this @aliok! I'm really excited about how many ideas the Knative team are putting together!

We do need a maintainer for each project -- they don't have to be the primary maintainer, but they should be involved (at least in an oversight capacity). Could you add them to this PR, [and the previous one](https://github.com/cncf/mentoring/pull/779/files#r1088070416)? We also need email addresses for mentors for when it comes time to add folks to the system.

File: summerofcode/2023.md [hunk 1]
```diff
@@ -39,7 +42,7 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
 - Description: We see more users asking for complete isolation in Knative Eventing deployments. While there are Knative Eventing components that support isolated data planes, we are interested in to see isolated control planes as well. There were discussions about this in the community before and many of the asks were left unadressed. However, we have tools that support multitenancy today, such as [kcp](https://github.com/kcp-dev/kcp). We see this project as an experiment.
 - Expected outcome: Finding issues blocking multiple control planes, and possibly fixing them.
 - Recommended Skills: Golang, Kubernetes, Knative, Kubernetes Controllers
-- Mentor(s):  Ali Ok @aliok (aliok AT redhat DOT com)
+- Mentor(s):  Ali Ok @aliok (aliok AT redhat DOT com), Christoph Stäbler @creydr (cstaebler AT redhat DOT com)
```
Comment (aliok) [inline summerofcode/2023.md:45]: Adding a co-mentor here

**Comment by aliok** [comment]
/unhold

#808 is merged.

@nate-double-u I've updated this PR with emails of the mentors. I've asked the maintainers listed in [project-maintainers.csv](https://github.com/cncf/foundation/blob/main/project-maintainers.csv#L1042-L1051) to approve these ideas here.

**Review by zroubalik — APPROVED** [review]
LGTM

**Comment by aliok** [comment]
For future reference, we're good with this PR based on [this](https://cloud-native.slack.com/archives/C04LMU33V1S/p1675272151365049) Slack discussion.

As Knative's governance varies from other projects, we cannot have every person in [project-maintainers.csv](https://github.com/cncf/foundation/blob/main/project-maintainers.csv#L1042-L1051) file although the people in this PR are the experts in their subprojects / workingGroups. Instead, we asked for an approval from a person in that file (@zroubalik LGTM'ed above).

**Review by nate-double-u — APPROVED** [review]
Thanks for this @aliok & Knative team!

**Pull request merged**